docs/core/model_training.md

```diff
@@ -0,0 +1,341 @@
+# Training and evaluating custom models on OCP datasets
+
+## Getting Started
+
+The [Open Catalyst Project](https://opencatalystproject.org/) consists of three distinct tasks:
+- Initial Structure to Relaxed Energy prediction (IS2RE)
+- Structure to Energy and Forces (S2EF)
+- Initial Structure to Relaxed Structure (IS2RS)
+
+This document is a tutorial for training and evaluating models
+for each of these tasks as well as generating submission files for the
+[evaluation server hosted on EvalAI](https://eval.ai/web/challenges/challenge-page/712/overview).
+
+`main.py` serves as the entry point to run any task. This script requires two command line
+arguments at a minimum:
+* `--mode MODE`: MODE can be `train`, `predict` or `run-relaxations` to train a model, make predictions
+using an existing model, or run machine learning based relaxations using an existing model, respectively.
+* `--config-yml PATH`: PATH is the path to a YAML configuration file. We use YAML files to supply all
+parameters to the script. The `configs` directory contains a number of example config files.
+
+Running `main.py` directly runs the model on a single CPU or GPU if one is available:
+```
+python main.py --mode train --config-yml configs/TASK/SIZE/MODEL/MODEL.yml
+```
+If you have multiple
+GPUs, you can use distributed data parallel training by running:
+```
+python -u -m torch.distributed.launch --nproc_per_node=8 main.py --distributed --num-gpus 8 [...]
```

Comment:
@lbluque per our discussion last time, this is getting deprecated - https://pytorch.org/docs/stable/distributed.html#launch-utility. Double check that the pytorch version of the ocp package is pre/post this and update accordingly.

Reply:
I tested runs with `torchrun` using #654, so we can keep as is right now, and I will update this in #654 once this is merged onto #623